test/unit_tests/braket/circuits/test_box_drawing_circuit_diagram.py

```diff
@@ -0,0 +1,986 @@
+        "      ┌─┴─┐   │         ┌────────────────┐ ┌─────┴─────┐ ",
+        "q2 : ─┤ X ├───┼─────────┤ Expectation(Y) ├─┤ Sample(Y) ├─",
+        "      └───┘   │         └────────────────┘ └─────┬─────┘ ",
+        "            ┌─┴─┐                          ┌─────┴─────┐ ",
+        "q3 : ───────┤ X ├──────────────────────────┤ Sample(Y) ├─",
+        "            └───┘                          └───────────┘ ",
+        "T  : │     0     │  1  │          Result Types          │",
+    )
+    _assert_correct_diagram(circ, expected)
+
+
+def test_multiple_result_types_with_state_vector_amplitude():
+    circ = (
+        Circuit()
+        .cnot(0, 2)
+        .cnot(1, 3)
+        .h(0)
+        .variance(observable=Observable.Y(), target=0)
+        .expectation(observable=Observable.Y(), target=3)
+        .expectation(observable=Observable.Hermitian(np.array([[1.0, 0.0], [0.0, 1.0]])), target=1)
+        .amplitude(["0001"])
+        .state_vector()
+    )
+    expected = (
+        "T  : │     0     │  1  │       Result Types       │",
+        "                  ┌───┐       ┌─────────────┐      ",
+        "q0 : ───●─────────┤ H ├───────┤ Variance(Y) ├──────",
+        "        │         └───┘       └─────────────┘      ",
+        "        │               ┌────────────────────────┐ ",
+        "q1 : ───┼─────●─────────┤ Expectation(Hermitian) ├─",
+        "        │     │         └────────────────────────┘ ",
+        "      ┌─┴─┐   │                                    ",
+        "q2 : ─┤ X ├───┼────────────────────────────────────",
+        "      └───┘   │                                    ",
+        "            ┌─┴─┐           ┌────────────────┐     ",
+        "q3 : ───────┤ X ├───────────┤ Expectation(Y) ├─────",
+        "            └───┘           └────────────────┘     ",
+        "T  : │     0     │  1  │       Result Types       │",
+        "",
+        "Additional result types: Amplitude(0001), StateVector",
+    )
+    _assert_correct_diagram(circ, expected)
+
+
+def test_multiple_result_types_with_custom_hermitian_ascii_symbol():
+    herm_matrix = (Observable.Y() @ Observable.Z()).to_matrix()
+    circ = (
+        Circuit()
+        .cnot(0, 2)
+        .cnot(1, 3)
+        .h(0)
+        .variance(observable=Observable.Y(), target=0)
+        .expectation(observable=Observable.Y(), target=3)
+        .expectation(
+            observable=Observable.Hermitian(
+                matrix=herm_matrix,
+                display_name="MyHerm",
+            ),
+            target=[1, 2],
+        )
+    )
+    expected = (
+        "T  : │     0     │  1  │     Result Types      │",
+        "                  ┌───┐     ┌─────────────┐     ",
+        "q0 : ───●─────────┤ H ├─────┤ Variance(Y) ├─────",
+        "        │         └───┘     └─────────────┘     ",
+        "        │               ┌─────────────────────┐ ",
+        "q1 : ───┼─────●─────────┤ Expectation(MyHerm) ├─",
```

Comment:
This doesn't support `Program`s with OpenQASM source, is that right? Asking here because I wonder how `measure q;` would look.